utils/led.c

```diff
@@ -0,0 +1,113 @@
+/*
+ * led.c
+ *
+ *  Created on: 29-Jan-2021
+ *      Author: Sonal Shrivastava
+ */
+
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+#include "drivers/mss_timer/mss_timer.h"
+#include "drivers_config/sys_config/sys_config_mss_clocks.h"
+#include "mss_gpio/mss_gpio.h"
+#include "led_error.h"
+#include "led.h"
+#include "CMSIS/system_m2sxxx.h"
+
+
+/*
+ * LEDs masks used to switch on/off LED through GPIOs.
+ */
+#define LEDS_MASK   (uint32_t)0xAAAAAAAA
+
+
+volatile uint32_t g_gpio_pattern = LEDS_MASK;
+static uint8_t num_of_blinks, count;
```

Comment:
I would see the `num_of_blinks` being renamed to a `num_led_toggles` and assigned to `blinks * 2` in the function... but if it makes more sense to you like this, it is perfectly readable as is!

Reply:
Yes I guess. But if it is okay with you I would like to leave it as is. But your word is final. Let me know.

Reply:
Perfectly fine like this!